File: src/Services/WareHouse/WareHouse.Domain/Entity/Outward.cs
﻿using System;
using System.Collections.Generic;

#nullable disable

namespace WareHouse.Domain.Entity
{
    public partial class Outward:BaseEntity
    {
        public Outward()
        {
            OutwardDetails = new HashSet<OutwardDetail>();
        }

        public string VoucherCode { get; set; }
        public DateTime VoucherDate { get; set; }
        public string WareHouseId { get; set; }
        public string ToWareHouseId { get; set; }
        public string Deliver { get; set; }
        public string Receiver { get; set; }
        public string Reason { get; set; }
        public string ReasonDescription { get; set; }
        public string Description { get; set; }
        public string Reference { get; set; }
        public DateTime CreatedDate { get; set; }
        public string CreatedBy { get; set; }
        public DateTime ModifiedDate { get; set; }
        public string ModifiedBy { get; set; }

        public virtual WareHouse ToWareHouse { get; set; }
        public virtual WareHouse WareHouse { get; set; }
        public virtual ICollection<OutwardDetail> OutwardDetails { get; set; }
    }
}

File: src/WebClient/WareHouse/src/app/entity/Outward.ts
import { BaseEntity } from "./BaseEntity";
import { Inward } from "./Inward";
import { OutwardDetail } from "./OutwardDetail";
import { WareHouse } from "./WareHouse";
import { WareHouseItem } from "./WareHouseItem";

export interface Outward extends BaseEntity {
    voucher:string;
    voucherCode: string;
    voucherDate: string;
    wareHouseId: string;
    toWareHouseId: string;
    deliver: string;
    receiver: string;
    reason: number;
    reasonDescription: string;
    description: string;
    reference: string;
    createdDate: string;
    createdBy: string;
    modifiedDate: string;
    modifiedBy: string;
    deliverPhone: string;
    deliverAddress: string;
    deliverDepartment: string;
    receiverPhone: string;
    receiverAddress: string;
    receiverDepartment: string;
    toWareHouse: WareHouse;
    wareHouse: WareHouse;
    outwardDetails: OutwardDetail[];
}